app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -137,6 +145,9 @@ class MainActivity : AppCompatActivity(), ReplayModeFragment.OnSelectModeItemLis
         override fun onCameraUpdated(camera: com.mapbox.vision.mobile.core.models.Camera) {
             calibrationProgress = camera.calibrationProgress
             runOnUiThread {
+                if (visionManagerMode == Replay && playbackTracked) {
```

Comment:
maybe create function `ReturnIfReplay`
inline fun returnIfReplay(){
   if (visionManagerMode == Replay && playbackTracked) {    
       return
    }
}

Reply:
I've thought about it but decided to avoid.
It is logic to manage execution flow outside the current call, have a feeling that better to use explicit code (some bad associations and experience with C++ macro)

Reply:
but it repeat several times
also you can do smth like:
`runOnUIIfReplay(block: lambda)`

Reply:
Seems better, fixed with this idea